### ISSUE TITLE
[Feature]: Create OpenDevin sandbox image for developing OpenDevin

### ISSUE THREAD

**Issue by neubig** [OpenDevin/OpenDevin#3185]
**What problem or use case are you trying to solve?**

We would like to use OpenDevin to solve OpenDevin issues. However, OpenDevin's default sandbox does not include much of the software that is used in developing OpenDevin, such as:
- poetry
- node
- docker

**Describe the UX of the solution you'd like**

It would be nice if we had an image that included all of the basic prerequisites for developing OpenDevin.

**Do you have thoughts on the technical implementation?**

Create a Dockerfile, and push an image to the OpenDevin docker hub.

**Comment by neubig** [comment]
Here's a start, but it doesn't handle the tricky part -- docker.

```dockerfile
FROM python:3.11-buster

# Run needed updates
RUN apt-get update && apt-get install -y

# Install node
RUN apt-get install -y nodejs

# Install poetry
RUN pip install poetry
```

**Comment by enyst** [comment]
I made a custom opendevin sandbox image in the past, with the backend dependencies pre-installed. It was a bit hacky though, I had some issue with ssh or non-root user, I don't remember exactly. I think it worked root-only.
I am still to check the new runtime, I'd guess it will be easier with it.

**Comment by tobitege** [comment]
I looked around custom docker images the other day and found this one:
https://hub.docker.com/r/nikolaik/python-nodejs?uuid=F0C119AE-6A9E-482C-A442-68C8BE1491EA
    Node.js: 22.x
    npm: 10.x
    yarn: stable
    Python: latest
    pip: latest
    pipenv: latest
    poetry: latest
    uv: latest
From the site you can just pick the one with the Python version you want.

**Comment by neubig** [comment]
Hey @tobitege , that actually worked perfectly, thanks so much for sharing!
I think we could add this to the [doc](https://github.com/OpenDevin/OpenDevin/blob/5f177b6f883d6b959e27107e6416cff5248588ff/docs/modules/usage/custom_sandbox_guide.md?plain=1#L5), so I'll leave this open for now.

**Comment by neubig** [comment]
Fixed by #3197
